### ISSUE TITLE
"Invalid modulus" while trying to login using protonvpn-cli login pvpn_username

### ISSUE THREAD

**Issue by sergiuszm** [ProtonVPN/linux-cli#19]
I use the newest release of ProtonVPN linux app with ProtonVPN CLI v3.7.1 (protonvpn-nm-lib v3.3.1; proton-client v0.5.1) on Linux Mint 20.1.

I am unable to use: `protonvpn-cli login pvpn_username`. Each time I type the correct username and correct password I get the following python error: `ValueError: Invalid modulus`. I attached logs that might be helpful in identifying the issue.

I have tried to troubleshot the issue myself but I do not have any more ideas. Any help would be greatly appreciated. 

[protonvpn.log](https://github.com/ProtonVPN/linux-cli/files/6721850/protonvpn.log)
[protonvpn-cli.log](https://github.com/ProtonVPN/linux-cli/files/6721851/protonvpn-cli.log)

**Comment by thiswillbeyourgithub** [comment]
Same here, I took a look at /var/lib/python3/dist-packages/proton/api.py and the line 136 : 
```
        verified = self.__gnupg.decrypt(armored_modulus)"
```
seems to be part of the issue. The armored modulus is sent to `gnupg.GPG().decrypt()` and the output is indeed indalid so the check fails.

So, assuming my login prompt is correct, the wrong "amored modulus" variable is sent to gnupg and the decryption fails.

(I have some python knowledge but don't count on my analysis for anything related to the actual encryption. I am on ubuntu 18.04 if that matters.)

**Comment by calexandru2018** [comment]
Hey @sergiuszm 
Just to be sure, were you introducing your protonmail username and password and not your openvpn data ?

**Comment by sergiuszm** [comment]
@calexandru2018 Yes, the input I have used was my protonmail username and password.

**Comment by thiswillbeyourgithub** [comment]
Same for me. But I tried both, and I actually tried "fake" username and got the same results (for example I removed the last character of my username)

**Comment by calexandru2018** [comment]
@sergiuszm could you try changing the password to the same you have now (instead of to a new one), then tre logging in again.

**Comment by sergiuszm** [comment]
@calexandru2018 Unfortunately, still the same error. Even with some fake data the error remains the same.

Might enabled 2FA be an issue here?

**Comment by calexandru2018** [comment]
@sergiuszm thank you for the quick test. 2FA shouldn't be a problem as I also have it enabled. Could you please contact our support at: https://protonvpn.com/support-form

Also provide the link of this issue so they can see what we've attempted please.

**Comment by sergiuszm** [comment]
@calexandru2018 Thanks. I have send a new request to the support.

**Comment by thiswillbeyourgithub** [comment]
Btw I don't have 2FA enabled and still have the same issue.

I didn't try to change the password to the same thing.

Note that I was able to use the community protonvpn cli utility for a few months without issue. I just uninstalled it right now to see if the latest release worked but apparently not.

**Comment by calexandru2018** [comment]
@thiswillbeyourgithub the community client works because it uses a completely different implementation.

**Comment by sergiuszm** [comment]
I have tried multiple different things with the help of ProtonVPN support but at the end I managed to fix the issue myself. I had to debug the code live in order to get the error: "GNUGP: No public key found...". So the code was not able to store ProtonVPN public key to decode the login information from the API.

The solution for me was to check and fix permission to the files in /home/user/.gnupg/ Some of the files were owned by root which was incorrect.

**Comment by thiswillbeyourgithub** [comment]
Thanks a lot @sergiuszm 

I was able to fix the issue and get a successful login by following the commands indicated there : https://superuser.com/questions/954509/what-are-the-correct-permissions-for-the-gnupg-enclosing-folder-gpg-warning

Apparently my gnupg settings were incorrect too. I would have never found out without you, thanks a lot!

**Comment by calexandru2018** [comment]
Interesting, was not aware of such possible situation. Thank you for sharing your solution with the community!

**Comment by cyberflamingo** [comment]
In addition to what has been said about the `~/.gnupg` folder and its permission, it seems the ProtonVPN app does not work if `$GNUPGHOME` is set to anything else than `~/.gnupg`.

I had it set to `$XDG_DATA_HOME/gnupg` in order to de-clutter my `$HOME` and it was the reason I could not launch the app or connect with the CLI anymore.

For future reference, use `echo $GNUPGHOME` to see if the variable is set. If nothing is output, no more action is required. If something is output and it is not `~/.gnupg` or equivalent, make sure to unset the variable.

The variable should be set in `.profile` for bash, `.zshenv` for zsh or `config.fish` for fish.

I have not check the source code yet but a small patch (the like of _if variable `$GNUPGHOME` is set, use it, otherwise check `~/.gnupg`..._) should be enough to counter this problem.
